python/ray/train/tests/test_tune.py

```diff
@@ -227,82 +224,6 @@ def train_func():
     assert len(trial_dfs[0]["training_iteration"]) == 4
 
 
-def test_tune_error_legacy(ray_start_4_cpus):
-    def train_func(config):
-        raise RuntimeError("Error in training function!")
-
-    trainer = Trainer(TestConfig(), num_workers=1)
-    TestTrainable = trainer.to_tune_trainable(train_func)
-
-    with pytest.raises(TuneError):
-        tune.run(TestTrainable)
-
-
-def test_tune_checkpoint_legacy(ray_start_4_cpus):
-    def train_func():
-        for i in range(10):
-            train.report(test=i)
-        train.save_checkpoint(hello="world")
-
-    trainer = Trainer(TestConfig(), num_workers=1)
-    TestTrainable = trainer.to_tune_trainable(train_func)
-
-    [trial] = tune.run(TestTrainable).trials
-    checkpoint_path = trial.checkpoint.dir_or_data
-    assert os.path.exists(checkpoint_path)
-    checkpoint = Checkpoint.from_directory(checkpoint_path).to_dict()
-    assert checkpoint["hello"] == "world"
-
-
-def test_reuse_checkpoint_legacy(ray_start_4_cpus):
-    def train_func(config):
-        itr = 0
-        ckpt = train.load_checkpoint()
-        if ckpt is not None:
-            itr = ckpt["iter"] + 1
-
-        for i in range(itr, config["max_iter"]):
-            train.save_checkpoint(iter=i)
-            train.report(test=i, training_iteration=i)
-
-    trainer = Trainer(TestConfig(), num_workers=1)
-    TestTrainable = trainer.to_tune_trainable(train_func)
-
-    [trial] = tune.run(TestTrainable, config={"max_iter": 5}).trials
-    checkpoint_path = trial.checkpoint.dir_or_data
-    checkpoint = Checkpoint.from_directory(checkpoint_path).to_dict()
-    assert checkpoint["iter"] == 4
-    analysis = tune.run(TestTrainable, config={"max_iter": 10}, restore=checkpoint_path)
-    trial_dfs = list(analysis.trial_dataframes.values())
-    assert len(trial_dfs[0]["training_iteration"]) == 5
-
-
-def test_retry_legacy(ray_start_4_cpus):
-    def train_func():
-        ckpt = train.load_checkpoint()
-        restored = bool(ckpt)  # Does a previous checkpoint exist?
-        itr = 0
-        if ckpt:
-            itr = ckpt["iter"] + 1
-
-        for i in range(itr, 4):
-            if i == 2 and not restored:
-                raise Exception("try to fail me")
-            train.save_checkpoint(iter=i)
-            train.report(test=i, training_iteration=i)
-
-    trainer = Trainer(TestConfig(), num_workers=1)
-    TestTrainable = trainer.to_tune_trainable(train_func)
-
-    analysis = tune.run(TestTrainable, max_failures=3)
-    checkpoint_path = analysis.trials[0].checkpoint.dir_or_data
-    checkpoint = Checkpoint.from_directory(checkpoint_path).to_dict()
-    assert checkpoint["iter"] == 3
-
-    trial_dfs = list(analysis.trial_dataframes.values())
-    assert len(trial_dfs[0]["training_iteration"]) == 4
-
-
```

Comment:
just to confirm, these are all covered in other tests right?

Reply:
Yes, we do